packages/jest-runner/src/types.ts

```diff
@@ -63,7 +63,6 @@ export type WatcherState = {
 };
 export interface TestWatcher extends EventEmitter {
   state: WatcherState;
-  new ({isWatchMode}: {isWatchMode: boolean}): TestWatcher;
```

Comment:
I believe this type was actually wrong. It implies you can do `new testWatcher({isWatchMode:true})`, rather than the intent of `new TestWatcher({isWatchMode:true})`. Interfaces don't have constructors.

Reply:
Some of the examples here have constructors? https://www.typescriptlang.org/docs/handbook/interfaces.html#difference-between-the-static-and-instance-sides-of-classes But it seems like we have to separate them out, or something - I might need to read up on it some more

Reply:
How would we correctly type this so TS knows we'll `new` the function/class they pass us with `{isWatchMode: boolean}`?

Reply:
Something might have changed? This was my script to try it out
```ts
interface Hey {
  new(foo: number): Hey;
  woo: number;
}

function testFunction(hey: Hey) {
  new hey(123);
}
```

Reply:
Ah, yes nothing changed. This is consistent with the docs link you sent. Notice they are defining the interface for the constructor itself, not the Clock.
```ts
interface ClockConstructor {
  new (hour: number, minute: number): ClockInterface;
}

interface ClockInterface {
  tick(): void;
}
```
We could add a TestWatcherConstructor interface, but that doesn't seem to represent anything public.

Reply:
Yeah, it's definitely wrong now, and just deleting is an improvement. The docs say something about the static part (like the constructor) needing to be typed separately from the "instance"  functions.

Reply:
Should we make it an abstract class? That's what I would do in Java, and maybe correct here as well?

Reply:
An interface looks right to me. The TestWatcher constructor is private. Consumers of this type only ever see TestWatcher instances.

Reply:
The implementation lives here: https://github.com/facebook/jest/blob/c83051789cc60789f2c575374270236e3a428879/packages/jest-core/src/TestWatcher.ts

Maybe we should just move that to the `jest-runner` package rather than duplicating its interface in an `interface`? We even import it from core in a test, so seems reasonable: https://github.com/facebook/jest/blob/4a5bac1459ed14b53262935b0bc0d6d6613b2000/packages/jest-runner/src/__tests__/testRunner.test.ts#L9

Reply:
meh, it's using `export =` and already has a TODO it should be moved